### ISSUE TITLE
Cannot set same timezone in user profile as in admin settings

### ISSUE THREAD

**Issue by bdeshi** [dshanske/simple-location#219]
as the issue says, I cannot set any timezone in user profile that matches what's set in admin settings.

For example, suppose **Settings > General > Time zone** is set to _UTC-10_.
Now **User > Profile > Time zone** can not be saved as either _UTC-10_ or _Honolulu_

---
|component|version|
|-|-|
|Simple-Location|4.4.9|
|WordPress|5.8|
|PHP|7.4.22|
|MariaDB|10.5.11|

**Comment by dshanske** [comment]
I don't remember adding in a timezone field to the user profile... I'll look into this

**Comment by bdeshi** [comment]
@dshanske I'm talking about this form: https://github.com/dshanske/simple-location/blob/trunk/templates/loc-user-metabox.php
This shows up at the bottom of [/wp-admin/profile.php](#)

**Comment by dshanske** [comment]
I forgot I did that.... will be having a detailed look

**Comment by dshanske** [comment]
This is what happens when you have so many moving parts.. I remember proposing this feature, but I don't remember building it.

**Comment by dshanske** [comment]
What feature would you want with it? I originally wanted it for the user profile.

**Comment by bdeshi** [comment]
Well I have no problem with _where_ it goes, I'd be happy if the timezone field here just worked properly.

Let me first share a work-around to get both user profile and site-wide time zone to match:

Suppose my desired TZ is UTC+6.
1. Switch the site TZ  to anything other than UTC+6 and save.
2. Now Simple-Location's timezone can be set to UTC+6. Do that and save.
3. Now reset site TZ back to UTC+6.

Okay. I discovered this problem while investigating why my post's publication times were getting shifted around. For example, a post published at 9AM was showing its time as 3AM, after publication.

I confirmed that the site tz was UTC+6, then noticed there was no value set in simple-location's timezone box at user profile settings page. Thought I'd forgotten to configure this after installing the plugin,  so I set it to UTC+6 too.
This setting didn't actually stick though, but because there were no warnings, I didn't double check.

Anyway, now I'm writing another post, and set the publishing time manually to 30th August 20201, 9:00AM.
Now, you know, simple-location also shows a [location metabox at the post editor screen](https://github.com/dshanske/simple-location/blob/trunk/templates/loc-side-metabox.php). This one was showing UTC+6 timezone[1]. So I believed everything should be alright. but after publishing the post, its time changed to 3:00AM again.

After some head-scratching, I discovered and used the work-around described above to match site's and simple-location's timezone, and this time-shifting problem has disappeared.

---
[1]: The Location metabox at post editor actually shows the timezone from the sitewide config, this should be sourced from the author's user profile setting, shouldn't it?
